variables.tf

```diff
@@ -335,6 +335,24 @@ variable "key_vault_secrets_provider_enabled" {
   nullable    = false
 }
 
+variable "kms_enabled" {
+  type        = bool
+  description = "(Optional) Enable Azure KeyVault Key Management Service."
+  default     = false
+}
+
+variable "kms_key_vault_key_id" {
+  type        = string
+  description = "(Optional) Identifier of Azure Key Vault key. When Azure Key Vault key management service is enabled, this field is required and must be a valid key identifier."
+  default     = null
+}
+
+variable "kms_key_vault_network_access" {
+  type        = string
+  description = "(Optional) Network Access of Azure Key Vault. Possible values are: `Private` and `Public`. If not set, defaults to type `Public`."
+  default     = null
```

Comment:
Should the default value be `Public` here?

Reply:
Not sure if it is good practice to bubble defaults from the provider up to the module.
It's already bubbled up from go-sdk to terraform provider. So if it's set to `null`, it defaults to `Public` on the provider level.

But I am also fine do default it also here. :-)

Reply:
👍 will do